### PR TITLE
the default value must be set for an empty options parameter

### LINKED ISSUE #638: The default value must be set if an empty object is passed to subscribe/publish functions
https://github.com/mqttjs/MQTT.js/pull/626#issuecomment-308463305

JS libary cannnot refuse empty object as function parameter. So, the default value must be set if an empty object is specified.

* The default options for [publish](https://github.com/mqttjs/MQTT.js#publish) are  `{qos: 0, retain: false, dup: false}`
* The default options for [subscribe](https://github.com/mqttjs/MQTT.js#subscribe)) is `{qos: 0}`

### PR DESCRIPTION
**Pull request by yohei1126**
fixes #638

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -440,9 +440,8 @@ MqttClient.prototype.subscribe = function () {
     return this
   }
 
-  if (!opts) {
-    opts = { qos: 0 }
-  }
+  const defaultOpts = { qos: 0 }
```
Comment (mcollina) [inline lib/client.js]: this should be `var`.

Comment (yohei1126) [inline lib/client.js]: Thanks!

**Comment by yohei1126** [comment]
@mcollina Could you check why the builds on Node 4.x were failed? The tests did not start.
https://travis-ci.org/mqttjs/MQTT.js/builds/250386104

**Comment by yohei1126** [comment]
Oops. Is extend not available on Node 4.0?

```
Error: Cannot find module 'extend' from '/home/travis/build/mqttjs/MQTT.js/lib'
```

**Comment by mcollina** [comment]
You should use the xtend module, not xtend, that's what we use through the normal dependencies. extend is likely in one of the tree, but we do not depend it directly and so it fails on an old version of npm.

**Comment by yohei1126** [comment]
@mcollina Thanks. Please check it again.

**Comment by mcollina** [comment]
Perfect!

**Pull request merged**